datafusion/expr/src/expr.rs

```diff
@@ -1775,6 +1775,27 @@ impl Expr {
                 | Expr::SimilarTo(Like { expr, pattern, .. }) => {
                     rewrite_placeholder(pattern.as_mut(), expr.as_ref(), schema)?;
                 }
+                Expr::InSubquery(InSubquery {
+                    expr,
+                    subquery,
+                    negated: _,
+                }) => {
+                    let subquery_schema = subquery.subquery.schema();
+                    let fields = subquery_schema.fields();
+
+                    // only supports subquery with exactly 1 field
+                    if let [first_field] = &fields[..] {
```

Comment:
What happens if there is more than one field? Will it not rewrite any placeholders?

Reply:
The subquery should only return one column. https://github.com/apache/datafusion/blob/main/datafusion/sql/src/expr/subquery.rs#L120